lib/main.js

```diff
@@ -324,9 +356,11 @@ const Hydrogen = {
       // Note that destroy alone does not shut down a WSKernel
       kernel.shutdown();
       kernel.destroy();
-    } else if (command === "rename-kernel" && kernel.promptRename) {
+    } else if (command === "rename-kernel" && kernel.kernel.promptRename) {
+      // XXX(nikita): breaks abstraction
       // $FlowFixMe Will only be called if remote kernel
-      if (kernel instanceof WSKernel) kernel.promptRename();
+      // XXX(nikita): breaks abstraction
```

Comment:
I'm not sure what's the intention of this comment?
The use of `promptRename` is guarded by `if (kernel.kernel instanceof WSKernel)`.

Reply:
I'd write:
```js
if (command === "rename-kernel" && kernel.kernel instanceof WSKernel) {
    kernel.kernel.promptRename();
}
```